packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx

```diff
@@ -0,0 +1,292 @@
+import Observation from './common/observation/_Observation';
+import observationPropShape from './proptypes/observationPropTypes';
+import ResultError from './common/other/_ResultError';
+import NoData from './common/other/_KnownNoData';
+import { isEmpty, checkIsStatusInError, ConditionalWrapper } from './common/utils';
+import styles from './ClinicalResult.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Result Object with the clinical result data.
+   */
+  resultData: PropTypes.shape({
+    /**
+     *  Blood Pressure grouped result id
+     */
+    id: PropTypes.string,
+    /**
+     *  Systolic Result for blood pressure.
+     */
+    systolic: observationPropShape,
+    /**
+     *  Diastolic Result for blood pressure.
+     */
+    diastolic: observationPropShape,
+  }),
+  /**
+   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
+   */
+  hideUnit: PropTypes.bool,
+  /**
+   * Whether or not the text should be truncated in display. Restricts clinical result details each to one line.
+   */
+  isTruncated: PropTypes.bool,
+  /**
+   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
+   */
+  hasResultError: PropTypes.bool,
+  /**
+   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
+   */
+  hasResultNoData: PropTypes.bool,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+};
+
+const defaultProps = {
+  resultData: {},
+  hideUnit: false,
+  isTruncated: false,
+  hasResultError: false,
+  hasResultNoData: false,
+};
+
+const ClinicalResultBloodPressure = (props) => {
+  const {
+    resultData,
+    hideUnit,
+    isTruncated,
+    hasResultError,
+    hasResultNoData,
+    intl,
+    ...customProps
+  } = props;
+
+  let clinicalResultBloodPressureDisplay = null;
+
+  if (hasResultError) {
+    clinicalResultBloodPressureDisplay = <ResultError />;
+  } else if (hasResultNoData) {
+    clinicalResultBloodPressureDisplay = <NoData />;
+  } else {
+    const CompareTemplate = (s = null, d = null) => ({
+      systolic: s,
+      diastolic: d,
+    });
+    const compareUnits = new CompareTemplate();
+    const compareConceptDisplays = new CompareTemplate();
+    const compareDatetimeDisplays = new CompareTemplate();
+    const compareStatusIsInError = new CompareTemplate();
+
+    let hasModifiedIcon = false;
+    let hasCommentIcon = false;
+    let hasUnverifiedIcon = false;
+
+    let iconGroupDisplayElement = null;
+    let conceptDisplayElement = null;
+    let datetimeDisplayElement = null;
+    const decoratedResultDisplay = [];
+
+    const hasSystolic = resultData.systolic;
+    const noDataSystolic = (hasSystolic && resultData.systolic.resultNoData === true);
```

Comment:
resultNoData is not a valid prop type for the observation props. This should be
```suggestion
    const noDataSystolic = (hasSystolic && !!resultData.systolic.result.value);
```

Reply:
And in reality, this should probably be a util helper to validate the result is a real/value number, not an empty string and is not undefined since the API allow for these.